### ISSUE TITLE
Using an (empty string) enum as the type of a primary key column

### ISSUE THREAD

**Issue by BTOdell** [typeorm/typeorm#3874]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[x] `0.2.15`

**Steps to reproduce or a small repository showing the problem:**

```ts
import {BaseEntity, Entity, PrimaryColumn, Column} from "typeorm";

enum Singleton {
    EMPTY = ""
}

/**
 * Global application settings.
 */
@Entity("settings")
export class Settings extends BaseEntity {

    @PrimaryColumn()
    readonly singleton: Singleton = Singleton.EMPTY;

    @Column()
    value!: string;

}
```

I have a "settings" table in my database that holds global application settings (one setting per column). In non-TypeORM projects, I've always used an enum primary key to enforce the existence of only one row. But TypeORM is not loading anything when I run:

```ts
Settings.find();
```

The array is empty (despite having a fully populated row in the database).

If I change the entity definition to:

```ts
import {BaseEntity, Entity, PrimaryColumn, Column} from "typeorm";

/**
 * Global application settings.
 */
@Entity("settings")
export class Settings extends BaseEntity {

    @PrimaryColumn()
    readonly singleton: number = 0;

    @Column()
    value!: string;

}
```

then it starts working (after updating the database schema).

**Comment by Kononnable** [comment]
It works if you don't set `EMPTY = ""` - set EMPTY as some string(with length>0) and you will be good.
Not sure why it's not working with empty string - probably because it's cast to false somewhere but it doesn't seem like a major bug.

**Comment by BTOdell** [comment]
I ran into this bug again now (a year later) and this time I did some initial investigation.
I have deduced that the bug exists in the code path of the `RawSqlResultsToEntityTransformer#transform` function: https://github.com/typeorm/typeorm/blob/master/src/query-builder/transformer/RawSqlResultsToEntityTransformer.ts#L39

The raw SQL results contain a single row (as expected) but this transformer fails to produce a valid entity from it.

I am going to continue digging through the code and eventually create a pull request with the fix.
